tailor_distro/manage/rosdistro_repo.py

```diff
@@ -0,0 +1,191 @@
+        self.internal_distro = get_distribution(self.internal_index, distro_name)
+
+        self.internal_distro_path = self.internal_index.distributions[distro_name]['distribution'][-1][len('file://'):]
+
+    def read_file(self, path):
+        path_obj = self.rosdistro_path / path
+        with path_obj.open() as f:
+            return f.read()
+
+    def write_file(self, path, new_contents, message=None):
+        path_obj = pathlib.Path(path)
+        path_obj.write_text(new_contents)
+
+    def get_branch_name(self):
+        rosdistro_repo = git.Repo(self.rosdistro_path)
+        return str(rosdistro_repo.active_branch)
+
+
+def get_distro_from_index(index, dist_name, type_='distribution'):
+    """
+       Portion of the logic from rosdistro.get_distribution
+    """
+    if dist_name not in index.distributions.keys():
+        valid_names = ', '.join(sorted(index.distributions.keys()))
+        raise RuntimeError("Unknown release: '{0}'. Valid release names are: {1}".format(dist_name, valid_names))
+    dist = index.distributions[dist_name]
+    if type_ not in dist.keys():
+        raise RuntimeError('unknown release type "%s"' % type_)
+
+    return dist[type_]
+
+
+class RemoteROSDistro(RepositoryBase):
+    def __init__(self, rosdistro_url, rosdistro_branch, distro_name):
+        repo_info = parse_repo_url(rosdistro_url)
+
+        if not repo_info:
+            raise RuntimeError(f'Cannot parse rosdistro_url {rosdistro_url}')
+
+        gh = get_github_client()
+        self.repo = gh.get_repo(repo_info['repo'])
+        self.path = repo_info['path']
+        self.branch = rosdistro_branch or repo_info['branch']
+
+        try:
+            self.internal_index = get_index(rosdistro_url)
+            self.internal_distro = get_distribution(self.internal_index, distro_name)
+        except HTTPError:
+            # Only needed for private github repositories
+            distro_folder = pathlib.Path(self.path).parent
+            self.internal_index = Index(self.read_yaml_file(self.path), str(distro_folder))
+
+            dist_url = get_distro_from_index(self.internal_index, distro_name)
+            if not isinstance(dist_url, list):
+                data = self.read_yaml_file(dist_url)
+            else:
+                data = []
+                for u in dist_url:
+                    data.append(self.read_yaml_file(str(u)))
+            self.internal_distro = create_distribution_file(distro_name, data)
+
+        self.internal_distro_path = self.internal_index.distributions[distro_name]['distribution'][-1]
+
+    def read_file(self, path):
+        git_file = self.repo.get_contents(path, self.branch)
+        return git_file.decoded_content.decode()
+
+    def write_file(self, path, new_contents, message=None, make_confirmation=True):
+        git_file = self.repo.get_contents(path, self.branch)
+        if make_confirmation:
+            # TODO(dlu): Should click be used for something?
```

Comment:
click is pretty good for interactive things (https://click.palletsprojects.com/en/5.x/api/#click.confirm)